File: frontend/src/core/models/alert.ts
export type AlertType = 'LOW_STOCK' | 'PO_OVERDUE' | 'MANUAL';
export interface Alert {
  id: number; alertType: AlertType;
  referenceId: number; referenceType: string;
  message: string; isRead: boolean; createdAt: string;
}

File: backend/alert-service/src/main/java/com/stockpro/alert/entity/Alert.java
package com.stockpro.alert.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "alerts")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Alert {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 30)
    private String alertType;

    private Long referenceId;
    private String referenceType;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String message;

    @Column(nullable = false)
    private Boolean isRead = false;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}